src/cpp/src/llm_pipeline.cpp

```diff
@@ -114,13 +138,43 @@ class LLMPipeline::LLMPipelineImpl {
         const ov::genai::Tokenizer& tokenizer,
         const std::string& device,
         const ov::AnyMap& plugin_config
-    );
+    ): 
+         m_model_runner{
+            ov::Core{}.compile_model(
+                model_path / "openvino_model.xml", 
+                device, 
+                plugin_config
+            ).create_infer_request()
+        },
+        m_tokenizer(tokenizer),
+        m_generation_config{from_config_json_if_exists(model_path)},
+        m_chat_template{chat_template_from_tokenizer_json_if_exists(model_path)}
+    {
+        // If eos_token_id was not provided, take value
+        if (m_generation_config.eos_token_id == -1)
+            m_generation_config.eos_token_id = m_tokenizer.get_eos_token_id();
+    }
 
     LLMPipelineImpl(
-        const std::filesystem::path& path, 
+        const std::filesystem::path& model_path, 
         const std::string& device, 
-        const ov::AnyMap& config
-    );
+        const ov::AnyMap& plugin_config
+    ): 
+         m_model_runner{
+            ov::Core{}.compile_model(
+                model_path / "openvino_model.xml", 
+                device, 
+                plugin_config
+            ).create_infer_request()
+        }, 
+        m_tokenizer(model_path.string()),
+        m_generation_config{from_config_json_if_exists(model_path)},
+        m_chat_template{chat_template_from_tokenizer_json_if_exists(model_path)}
+    {
+        // If eos_token_id was not provided, take value
+        if (m_generation_config.eos_token_id == -1)
+            m_generation_config.eos_token_id = m_tokenizer.get_eos_token_id();
+    }
```

Comment:
```suggestion
    ): LLMPipelineImpl{model_path, Tokenizer(model_path.string()), device,plugin_config} {}
```

Reply:
Applied. That's a great simplification, thanks x a lot!